atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -78,6 +78,8 @@
      * It is guaranteed that the {@link Map#keySet()} of the returned map has a corresponding element for each of the
      * input {@code rows}, even if there are rows where no columns match the predicate.
      *
+     * Batching visitables must be used strictly within the scope of the transaction.
```

Comment:
This is copy-and-pasted throughout, but I wonder if you maybe need to be a bit more explicit with what you mean at least once (maybe at the top):
```
Batching visitables, iterables, streams, and futures _must_ be used strictly within the scope of the transaction. Concretely, this means that results must be retrieved before exiting the transaction - in the case of streams and iterators, this means that you must collect the results before exiting; in the case of futures, you must await all asynchronous calls before returning from the transaction
```
Or something along those lines. I know what you mean here, but I have a feeling that users may not.
```

Reply:
Makes sense - yeah I've added this at the top. What you've written is very clear, so I've used that :)